cognite/client/data_classes/documents.py

```diff
@@ -36,7 +125,7 @@ class SourceFile(CogniteObject):
         size (int | None): The size of the file in bytes.
         asset_ids (list[int] | None): The ids of the assets related to this file.
         labels (list[Label | str | LabelDefinition] | None): A list of labels associated with this document's source file in CDF.
-        geo_location (GeoLocation | None): The geolocation of the source file.
+        geo_location (DocumentsGeoJsonGeometry | None): The geolocation of the source file.
```

Comment:
I feel like we are missing a class here: The old usage of `GeoLocation` should be replaced (for Document/-List/SourceFile etc.) by a new `DocumentsGeoLocation` class. This in turn should accept a collection of `DocumentGeoJsonGeometry`  (instead of `Geometry`) for `geometries`.

Reply:
The `geo_location` field _is_ a geometry object. It can be one of the standard geometry types, such as Point or Polygon, or the special geometry type `GeometryCollection`, which is the only case the `geometries` field should be used.

Reply:
Oh, does that mean the `GeoLocation` is redundant/should be removed?

Reply:
It looks like it is still used for other endpoints such as assets and files, so I don't think it can be removed.

In terms of the geojson spec, the `GeoLocation` object maps to the [Feature](https://datatracker.ietf.org/doc/html/rfc7946#section-3.2) type, which is some kind of a wrapper object around Geometries. The Documents API uses the [Geometry](https://datatracker.ietf.org/doc/html/rfc7946#section-3.2) objects directly.

An alternative thing I could have done instead of making the new `DocumentsGeoJsonGeometry` object, was to use the existing `Geometry` class directly. The existing `Geometry` class is a bit weird, because it explicitly does not support the `GeometryCollection` type, but it has a `geometries` field, which I don't think is actually used for anything at all. But I did not want to properly add `GeometryCollection` support to this class, since other services don't support it.

Another thing I probably could have done, was to make the `geo_location` field be a union type of `Geometry | GeometryCollection | None`, where `GeometryCollection` could be a new class that has the `geometries` field and type=`GeometryCollection`. Whether this will work or not, depends on the serialization/deserialization capabilities of the SDK, which I'm not so familiar with.